eng/docker/alpine.3.6/Dockerfile

```diff
@@ -18,10 +18,15 @@ RUN echo 'code_executor ALL=(ALL) NOPASSWD:ALL' >> /etc/sudoers
 RUN chmod -R a+rwx /usr/local
 RUN chmod -R a+rwx /home
 
-# Set user to the one we just created
-USER ${USER_ID}
-
 # Set working directory 
 ARG WORK_DIR
 WORKDIR ${WORK_DIR}
 
+# Set up Azure Artifacts credential provider
+RUN curl -O https://raw.githubusercontent.com/Microsoft/artifacts-credprovider/master/helpers/installcredprovider.sh
+RUN chmod +x ./installcredprovider.sh
```

Comment:
Surprisingly this wasn't working on Alpine and Ubuntu 16.04. The Linux user that execute this line is `root` and the installer will try to put the cred provider in the wrong folder: /root/.nuget instead of /home/code_executer/.nuget

Reply:
Interesting. Agreed it makes sense with those constraints, weird that the user can't write to its own home directory. Worth adding a comment with that description IMO.

Reply:
The curl would be able to put the file in the user directory if we `CD` to the user directory. `Curl` seems not to be able to save files other than the current directory.

Reply:
Are you talking about the curl in `curl -sSL https://raw.githubusercontent.com/Microsoft/artifacts-credprovider/master/helpers/installcredprovider.sh | bash`? That doesn't store the script as a file, just pipes it.

Reply:
Nope. I was referring to something like `curl <URL> -o /home/executor/.nuget/installcredprovider.sh`. That doesn't work, I've to `CD` to the target directory: https://stackoverflow.com/questions/16362402/save-file-to-specific-folder-with-curl-command

Reply:
What I'm saying is that you don't need to download `installcredprovider.sh` to a file. This approach won't hurt anything, I suppose, but that's more reason for a comment IMO.

Reply:
Added a comment explaining the reason.